### PR TITLE
Update ABI resolution scheme

### PR DESCRIPTION
**Pull request by finalstatics**
Update ABI resolution scheme

### CODE REVIEW

File: coinlib/src/main/java/com/keystone/coinlib/utils/AbiLoader.java [hunk 1]
```diff
@@ -1,54 +1,66 @@
...
-import java.io.BufferedReader;
 import java.io.File;
-import java.io.FileReader;
-import java.io.IOException;
 import java.lang.reflect.Method;
 import java.util.List;
 
 public class AbiLoader {
-    private static final String TAG = "AbiLoader";
+    public static final String INDEX_JSON_SDCARD_PATH = "contracts" + File.separator + "ethereum";
```
Comment (aaronisme) [inline coinlib/src/main/java/com/keystone/coinlib/utils/AbiLoader.java:18]: refactor the duplicate logic

**Review by aaronisme — CHANGES_REQUESTED** [review]

File: coinlib/src/main/java/com/keystone/coinlib/utils/AbiLoader.java [hunk 2]
```diff
@@ -1,54 +1,66 @@
...
+        } catch (SQLException e) {
             e.printStackTrace();
         } finally {
-            if (bfr != null) {
-                try {
-                    bfr.close();
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
+            db.close();
```
Comment (aaronisme) [inline coinlib/src/main/java/com/keystone/coinlib/utils/AbiLoader.java:33]: cursor should be close too？

File: coinlib/src/main/java/com/keystone/coinlib/coins/ETH/EthImpl.java [hunk 3]
```diff
@@ -134,8 +131,12 @@ public static JSONObject decodeRawTransaction(String txHex, Callback callback) {
                 abi = readAsset("abi/" + abiFile);
                 contractName = abiFile.replace(".json", "");
             } else {
-                abi = readAbiFromTFCard(rawTx.getTo(), callback);
-                contractName = contractNameFromTFCard(rawTx.getTo());
+                Contract dataFromTFCard = TFDBLoader.getDataFromTFCard(rawTx.getTo());
```
Comment (aaronisme) [inline coinlib/src/main/java/com/keystone/coinlib/coins/ETH/EthImpl.java:134]: TFDBLoader -> ContactExternalDbLoader

Comment (aaronisme) [reply coinlib/src/main/java/com/keystone/coinlib/coins/ETH/EthImpl.java:134]: dataFromTFCard -> ContractData

**Review by aaronisme — APPROVED** [review]
LGTM

File: app/src/main/java/com/keystone/cold/viewmodel/EthTxConfirmViewModel.java [hunk 4]
```diff
@@ -35,16 +35,15 @@
 import com.keystone.coinlib.interfaces.SignCallback;
 import com.keystone.coinlib.interfaces.Signer;
 import com.keystone.coinlib.path.CoinPath;
-import com.keystone.coinlib.utils.AbiLoader;
 import com.keystone.coinlib.utils.Coins;
+import com.keystone.coinlib.utils.ContactExternalDbLoader;
```
Comment (soralit) [inline app/src/main/java/com/keystone/cold/viewmodel/EthTxConfirmViewModel.java:39]: typo

**Review by aaronisme — APPROVED** [review]

**Pull request merged**